services/fluxmonitor.go

```diff
@@ -354,9 +353,46 @@ func (fm *FluxMonitor) getAdapterResponse(endpoint url.URL, requestData string)
 	return response.Price, json.Unmarshal(body, &response)
 }
 
-func (fm *FluxMonitor) poll() error {
+// PollAndGetAnswer() will poll if the latest answer is older than the polling interval
+// and then return the latest value
+func (fm *FluxMonitor) PollAndGetAnswer() (decimal.Decimal, error) {
```

Comment:
Maybe we can use only one function? the pollWithRetry and get rid of PollAndGetAnswer and 
ForceNewPoll?

The other 2 do not provide much functionality, duplicate logic a bit and add some complexity. 

In this case, mutex could be moved inside pollWithRetry?

Reply:
The reason for having the mutex here is if a process wanting to read the latest result while a different process is already polling, then the read process should be blocked until the polling is done and it can read the latest result. What happens if there's no lock for reading is that a process could actively be polling while a different process wants to read. Since it sees that the value is outdated, it will queue a new poll - which will process as soon as the previous poll is done. This means there would be two polls happening roughly at the same time.

Having multiple functions is to split up the different intentions (poll timer should always poll, while writing the value on-chain should get the latest value if possible, and poll if the result is outdated). Both of them should share the same lock, for the reason explained above.

Reply:
I see. Maybe what if we have 2 methods instead of 3, the `pollWithRetry`, with the mutex inside it + the `PollAndGetAnswer`, renamed to something like `ValidAnswer`) again using the same mutex(this is a good idea indeed, to check if polling is in progress) inside it?

But not have to call `fm.pollWithRetry()` at the end of `PollAndGetAnswer`(so there will be no need for `ForceNewPoll`), just call it subsequently if the check returns false, inside `checkAndSendJob`.

Reply:
I'm not sure if I fully understand what you mean - but if I understand correctly, there would still be a condition where two polls are initiated at the same time (`checkAndSendJob` sees that the result is outdated with lock, releases lock and queues a new poll with `PollAndGetAnswer`. Before that happens, poll/heartbeat timer could kick in and triggers a new poll, which would result in two polls right after each other).

Reply:
Hm, by the time `PollAndGetAnswer` releases the lock, it means that a previous poll has just ended and probably the `fm.latestResultTimestamp` would be valid for use, so it won't trigger a new poll anyway. 

The other case is that result is still empty(first time called) and a new poll will start immediately, pollTimer won't have time to get triggered meanwhile, it would be just a boolean check.

Created just a draft PR with what I was thinking -> https://github.com/smartcontractkit/external-initiator/pull/131/files

In any case, both cases work, probably shouldn't lose more time in these details, so if you feel like merging the current work, please do.

Reply:
Alright I see what you mean, we'll just roll with that I think